sdk/appconfiguration/azappconfiguration/configuration_setting.go

```diff
@@ -0,0 +1,69 @@
+//go:build go1.16
+// +build go1.16
+
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+package azappconfiguration
+
+import (
+	"time"
+
+	"sdk/appconfiguration/sdk/appconfiguration/azappconfiguration/internal/generated"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+)
+
+type ConfigurationSetting struct {
+	Key         *string
+	Value       *string
+	Label       *string
+	ContentType *string
+
+	etag         *azcore.ETag
+	tags         map[string]*string
+	lastModified *time.Time
+	isReadOnly   *bool
```

Comment:
These should be exported as well. A user might want to inspect etag, lastModified, and isReadOnly. Tags can be edited by the user directly and changed on a submit.

Reply:
There are getters below - `ETag()`, `Tags()`, `LastModified()`, and `IsReadOnly()`. But I see that some  SDK do allow them to be modified, so I guess there would be no harm in turning this to a struct with all public members. C# only has 4 out of 8 modifiable, while in Java SDK you can set 7 out of 8 (`setReadOnly` is private for customers).

Reply:
Is the configuration setting a model is input and output? (ie returned by the SDK and a parameter to the SDK)

Reply:
Yes, it it input and output. I guess one other advantage to use the same type, is that you can invoke Get(), and then Set() with the same variable. Other SDKs are also reusing the same abstraction for inputs and outputs. Plus, they have overloads, which allows them to have overload methods like `SetConfigurationSetting(key, value, label = default, options = default)` (in addition to ones that take ConfigurationSetting). We don't have overloads, so I think it is important to make interactions with `ConfigurationSetting` type syntactiaclly light, and struct would give us that - `SetConfigurationSetting(context, ConfigurationSetting{ Key: "name", Value: "value", Label: "label" }, null)`